File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/MainActivityBindingModule.java
package com.tohami.egyptinnovate.dagger.module;

import com.tohami.egyptinnovate.ui.news.details.view.NewsDetailsFragment;
import com.tohami.egyptinnovate.ui.news.list.view.NewsListFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module(includes = ViewModelModule.class)
public abstract class MainActivityBindingModule {

    @ContributesAndroidInjector
    abstract NewsListFragment provideNewsListFragment();

    @ContributesAndroidInjector
    abstract NewsDetailsFragment provideNewsDetailsFragment();
}

File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/MainActivityBindingModule.kt
package com.tohami.egyptinnovate.dagger.module

import com.tohami.egyptinnovate.ui.news.details.view.NewsDetailsFragment
import com.tohami.egyptinnovate.ui.news.list.view.NewsListFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module(includes = [ViewModelModule::class])
abstract class MainActivityBindingModule {

    @ContributesAndroidInjector
    internal abstract fun provideNewsListFragment(): NewsListFragment

    @ContributesAndroidInjector
    internal abstract fun provideNewsDetailsFragment(): NewsDetailsFragment
}
